torch_xla/distributed/xla_backend.py

```diff
@@ -0,0 +1,290 @@
+import distutils.util
+import os
+import torch
+import torch.distributed as dist
+import torch_xla
+import torch_xla.core.xla_model as xm
+import logging
+from torch._C._distributed_c10d import (
+    ProcessGroup,
+    Work,
+)
+
+
+def _create_xla_process_group(prefix_store, rank, size, timeout):
+  return ProcessGroupXla(prefix_store, rank, size, timeout)
+
+
+def _register_xla_backend():
+  dist.Backend.register_backend('xla', _create_xla_process_group)
+
+
+_register_xla_backend()
+
+
+class ProcessGroupXla(ProcessGroup):
+  '''ProcessGroup for XLA devices. See ProcessGroup for doc.
+
+    Here we are implementing only a Python subclass. For implementing a
+    C++/Python extension, see
```

Comment:
Do we have the C++ binding in a separate PR?

Reply:
not sure if aws plans to do that, it is also unclear to me how C++ binding is going to help